lib/govuk_publishing_components/presenters/select.rb

```diff
@@ -0,0 +1,42 @@
+module GovukPublishingComponents
+  module Presenters
+    class SelectHelper
+      attr_reader :options, :option_markup, :selected_option
+
+      def initialize(local_assigns)
+        @options = local_assigns[:options] if local_assigns[:options]
+        @option_markup = get_options
+      end
+
+    private
+
+      def get_options
+        return {} if @options.nil?
+
+        formatted_options = []
+
+        options.each do |option|
+          formatted_options << [
+            option[:text],
+            option[:value],
+            options_data_attribute(option[:data_attributes])
+          ]
+          @selected_option = option[:value] if option[:selected]
+        end
+
+        formatted_options
+      end
+
+      def options_data_attribute(attributes)
+        return {} if attributes.nil?
+
+        attrs = {}
+        attributes.each do |key, value|
+          key_name = "data-#{key.to_s.split('_').join('-')}"
```

Comment:
Is this needed? This method looks like it could be simplified into returning `{} if attributes.nil?` or just `attributes` if not.

Reply:
If you're referring to line 35's `to_s.split('_')` etc. then the reason for this is because data attributes are passed from the YML as keys with underscores, but data attributes need hyphens. Does that make sense or have I misunderstood your comment?

Reply:
I think the difference is on those other components we're using the rails tag generation stuff, which takes a data object and does the conversion. In this case we're using [options_for_select](https://apidock.com/rails/ActionView/Helpers/FormOptionsHelper/options_for_select), which appears to work slightly differently. Here's an example from the page:

```
options_for_select([["Dollar", "$", {class: "bold"}], ["Kroner", "DKK", {onclick: "alert('HI');"}]])
```

Firstly, if we just returned the attributes it doesn't put `data-` on the front of them, so we need to append that. And then unless we do the string replace `data_something` comes out as an attribute on the element called `data-something` - but if I pass through `data_something_else` it would come out as `data-something_else` which I don't think is right. 

Kind of frustrating that this rails helper doesn't seem to quite work the same as others, but I guess it's a bit unusual to want to put data attributes on an `option` element?